README.md

```diff
@@ -1,2 +1,105 @@
 # unified-user-data-format
+
 Specification for the unified user data format across the various "Tube" apps (NewPipe, FreeTube, Invidious, etc..)
+
+## Intro
+
+* This specification is a very early draft, and prone to regular changes
+* Unless otherwise specified, all fields are assumed to be `Strings`
+
+
+## Top level structure
+
+#### `"version"`
+
+The [semantic version](https://semver.org/) of the specification.
+
+Conditions for bumping major/minor/patch version: TBD
+Potential suffixes: `-draft`, TBD
+
+#### `"subscriptions"`
+
+An array of Subscription objects. See below for details.
+
+#### `"playlists"`
+
+An array of Playlist objects. See below for details.
+
+### Overview
+
+```javascript
+{
+	"version": "X.Y.Z[-draft/-rc]",
+	"subscriptions": Subscriptions[],
+	"playlists": Playlist[],
+}
+```
+
+## Subscription object
+
+#### > `"name"`
+
+The name of the subscribed channel.
+
+#### > `"url"`
+
+The URL used to access the user subscribed channel.
+
+#### > `"thumbnail"`
+
+The URL to the user/channel's thumbnail.
+
+### Example
+
+```javascript
+{
+  "name": "Linus Tech Tips",
+  "url": "https://www.youtube.com/channel/UCXuqSBlHAE6Xw-yeJA0Tunw",
+  "thumbnail": "https://yt3.ggpht.com/ytc/abcdef..."
+}
+```
+
+## Playlist object
+
+#### > `"name"`
+
+The title of the playlist.
+
+#### > `"visibility"`
+
+The visibility of the playlist.
+
+Accepted values: "public", "unlisted", "private"
+
+#### > `"description"`
+
+The description of the playlist.
+Format: plain text (? TBD)
+
+CAN be ommited; In this case, the parser MUST assume an empty `String`.
```

Comment:
nitpick: I feel null would be more appropriate if there's no description set/present

Reply:
I personally prefer omitting fields than null-ing them

Reply:
Json parsers in type/null-safe languages make it much easier/less likely to forget to check for null than an empty string.

Reply:
I also agree it's better to use `null` instead of omitting the field completely